controllers/release/adapter.go

```diff
@@ -288,28 +288,48 @@ func (a *adapter) EnsureReleaseProcessingIsTracked() (controller.OperationResult
 		}
 	}
 
-	// This condition can only be true if the call to registerProcessingStatus changed the state
-	if a.release.HasProcessingFinished() {
-		// At this point, the PipelineRun has finished, so it's safe to remove the finalizer and RoleBinding
-		roleBinding, err := a.loader.GetRoleBindingFromReleaseStatus(a.ctx, a.client, a.release)
-		if err != nil && !errors.IsNotFound(err) && !strings.Contains(err.Error(), "valid reference to a RoleBinding") {
-			return controller.RequeueWithError(err)
-		}
-		if roleBinding != nil {
-			err = a.client.Delete(a.ctx, roleBinding)
-			if err != nil {
-				return controller.RequeueWithError(err)
-			}
+	return controller.ContinueProcessing()
+}
+
+// EnsureReleaseProcessingResourcesAreCleanedUp is an operation that will ensure that the resources created for the Release
+// Processing step are cleaned up once processing is finished.
+func (a *adapter) EnsureReleaseProcessingResourcesAreCleanedUp() (controller.OperationResult, error) {
+	if !a.release.HasProcessingFinished() {
+		return controller.ContinueProcessing()
+	}
+
+	pipelineRun, err := a.loader.GetManagedReleasePipelineRun(a.ctx, a.client, a.release)
+	if err != nil && !errors.IsNotFound(err) {
+		return controller.RequeueWithError(err)
+	}
+
+	roleBinding, err := a.loader.GetRoleBindingFromReleaseStatus(a.ctx, a.client, a.release)
+	if err != nil && !errors.IsNotFound(err) && !strings.Contains(err.Error(), "valid reference to a RoleBinding") {
+		return controller.RequeueWithError(err)
+	}
+
+	return controller.RequeueOnErrorOrContinue(a.cleanupManagedPipelineRunResources(pipelineRun, roleBinding))
```

Comment:
Would you mind to use a name similar to the operation name? `cleanUpProcessingResources` for example?

Reply:
Much nicer, great idea. Updated